### PR TITLE
Added initial support for config files

### PR DESCRIPTION
**Pull request by roman-rezinkin**
This is the PR to introduce #10 into the static-site-generator.

### CODE REVIEW

File: ssg/ssg.py [hunk 1]
```diff
@@ -123,13 +124,32 @@ def output_to_file(file_name, html):
...
     args = parser.parse_args()
     
     if(os.path.isdir(OUTPUT_FOLDER)):
         shutil.rmtree(OUTPUT_FOLDER)
-
-    input = args.input
-    stylesheet = args.stylesheet
+    
+    # Check to see if Config File exists.
+    if args.config != None:
```
Comment (a-rokay) [inline ssg/ssg.py:134]: This can be simplified to "if args.config".

File: ssg/ssg.py [hunk 2]
```diff
@@ -132,18 +132,39 @@ def output_to_file(file_name, html):
...
+                else:
+                    print("\nInput File gathered from Config does not exist\n")
+            elif i == "stylesheet" or i == "s":
+                if os.path.isfile(data[i]):
                     stylesheet = data[i]
-                elif i == "lang" or i == "l":
-                    lang = data[i]
-        except (FileNotFoundError, RuntimeError, json.decoder.JSONDecodeError) as err:
-            print("\nError parsing Config File: {0}\n".format(err))
+                else:
```
Comment (a-rokay) [inline ssg/ssg.py:156]: I'd rather have it print nothing if something is ignored.

File: ssg/ssg.py [hunk 3]
```diff
@@ -132,18 +132,39 @@ def output_to_file(file_name, html):
...
+                exit()
+        # For each command from JSON config file, set local parameters
+        for i in data:
+            if i == "input" or i == "i":
+                if os.path.isfile(data[i]):
+                    input = data[i]
+                elif os.path.isdir(data[i]):
                     input = data[i]
-                elif i == "stylesheet" or i == "s":
+                else:
```
Comment (a-rokay) [inline ssg/ssg.py:151]: I'd rather have it print nothing if something is ignored.

File: ssg/ssg.py [hunk 4]
```diff
@@ -132,18 +132,39 @@ def output_to_file(file_name, html):
...
-            with open(args.config) as f:
+        with open(args.config) as f:
+            try:
                 data = json.load(f)
-            for i in data:
-                if i == "input" or i == "i":
+                if len(data) == 0:
+                    print("\nConfig File is empty!\n")
+                    exit()
+            except (FileNotFoundError, RuntimeError, json.decoder.JSONDecodeError) as err:
```
Comment (a-rokay) [inline ssg/ssg.py:141]: This should only catch JSONDecodeError.

File: ssg/ssg.py [hunk 5]
```diff
@@ -132,18 +132,39 @@ def output_to_file(file_name, html):
...
+                print("\nError parsing Config File: {0}\n".format(err))
+                exit()
+        # For each command from JSON config file, set local parameters
+        for i in data:
+            if i == "input" or i == "i":
+                if os.path.isfile(data[i]):
```
Comment (a-rokay) [inline ssg/ssg.py:147]: Redundant if/elif statement. Can just set "input = data[i]" after 146.

File: ssg/ssg.py [hunk 6]
```diff
@@ -132,18 +132,39 @@ def output_to_file(file_name, html):
...
+                    input = data[i]
+                elif os.path.isdir(data[i]):
                     input = data[i]
-                elif i == "stylesheet" or i == "s":
+                else:
+                    print("\nInput File gathered from Config does not exist\n")
+            elif i == "stylesheet" or i == "s":
+                if os.path.isfile(data[i]):
```
Comment (a-rokay) [inline ssg/ssg.py:154]: Redundant if statement.

File: ssg/ssg.py [hunk 7]
```diff
@@ -132,18 +132,39 @@ def output_to_file(file_name, html):
...
-        except (FileNotFoundError, RuntimeError, json.decoder.JSONDecodeError) as err:
-            print("\nError parsing Config File: {0}\n".format(err))
+                else:
+                    print("\nStylesheet gathered from Config does not exist\n")
+            elif i == "lang" or i == "l":
+                lang = data[i]
+        if input == None:
+            print("No input file specified")
+            exit()
+        try:
```
Comment (a-rokay) [inline ssg/ssg.py:163]: Sets both lang and stylesheet to default if one is missing.

**Review by a-rokay — COMMENTED** [review]
I've left some comments on your code

File: ssg/ssg.py (hunk 1 above)
Comment (roman-rezinkin) [reply ssg/ssg.py:134]: Thank you, I have address it in my latest commit.

**Comment by a-rokay** [comment]
Can you also put all of that logic in a function to clean it up a bit?

**Comment by a-rokay** [comment]
Looks good. Thanks!

**Pull request merged**